flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/RestartBackoffTimeStrategyFactoryLoader.java

```diff
@@ -0,0 +1,393 @@
+		final Configuration adaptedConfiguration) throws Exception {
+
+		final String restartStrategyName = clusterConfiguration.getString(ConfigConstants.RESTART_STRATEGY, null);
+		if (restartStrategyName == null) {
+			return tryAdaptingClusterConfigurationOfDeprecatedValues(clusterConfiguration, adaptedConfiguration);
+		}
+
+		switch (restartStrategyName.toLowerCase()) {
+			case "none":
+			case "off":
+			case "disable":
+				setNoRestartStrategyToConfiguration(adaptedConfiguration);
+				return true;
+			case "fixeddelay":
+			case "fixed-delay":
+				adaptClusterConfigurationOfFixedDelayRestartStrategy(
+					clusterConfiguration,
+					adaptedConfiguration);
+				return true;
+			case "failurerate":
+			case "failure-rate":
+				adaptClusterConfigurationOfFailureRateRestartStrategy(
+					clusterConfiguration,
+					adaptedConfiguration);
+				return true;
+			default:
+				return false;
+		}
+	}
+
+	private static boolean tryAdaptingClusterConfigurationOfDeprecatedValues(
+		final Configuration clusterConfiguration,
+		final Configuration adaptedConfiguration) throws Exception {
+
+		// support deprecated ConfigConstants values
+		final int maxAttempts = clusterConfiguration.getInteger(
+			ConfigConstants.RESTART_STRATEGY_FIXED_DELAY_ATTEMPTS,
+			ConfigConstants.DEFAULT_EXECUTION_RETRIES);
+
+		final Duration heartbeatPause = parseDurationFromConfig(
+			clusterConfiguration,
+			AkkaOptions.WATCH_HEARTBEAT_PAUSE);
+		final long delay = parseDurationFromConfig(
+			clusterConfiguration,
+			ConfigConstants.RESTART_STRATEGY_FIXED_DELAY_DELAY.key(),
+			heartbeatPause.toString()).toMillis();
+
+		if (maxAttempts > 0 && delay >= 0) {
+			setFixedDelayRestartStrategyToConfiguration(adaptedConfiguration, delay, maxAttempts);
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	private static void adaptClusterConfigurationOfFixedDelayRestartStrategy(
+		final Configuration clusterConfiguration,
+		final Configuration adaptedConfiguration) throws Exception {
+
+		final int maxAttempts = clusterConfiguration.getInteger(
+			ConfigConstants.RESTART_STRATEGY_FIXED_DELAY_ATTEMPTS,
+			1);
+
+		final long delay = parseDurationFromConfig(
+			clusterConfiguration,
+			ConfigConstants.RESTART_STRATEGY_FIXED_DELAY_DELAY).toMillis();
+
+		setFixedDelayRestartStrategyToConfiguration(adaptedConfiguration, delay, maxAttempts);
+	}
+
+	private static void adaptClusterConfigurationOfFailureRateRestartStrategy(
+		final Configuration clusterConfiguration,
+		final Configuration adaptedConfiguration) throws Exception {
+
+		final int maxFailuresPerInterval = clusterConfiguration.getInteger(
+			ConfigConstants.RESTART_STRATEGY_FAILURE_RATE_MAX_FAILURES_PER_INTERVAL,
+			1);
+
+		final long failuresInterval = parseDurationFromConfig(
+			clusterConfiguration,
+			ConfigConstants.RESTART_STRATEGY_FAILURE_RATE_FAILURE_RATE_INTERVAL,
+			Duration.apply(1, TimeUnit.MINUTES).toString()).toMillis();
+
+		final Duration heartbeatPause = parseDurationFromConfig(
```

Comment:
In legacy FailureRateRestartStrategy it uses `AkkaOptions.WATCH_HEARTBEAT_INTERVAL` as the default delay. But I think `AkkaOptions.WATCH_HEARTBEAT_PAUSE` should be the proper one.